pkg/apiroutes/templates.go

```diff
@@ -355,3 +357,23 @@ func BatchPatchTemplateRepos(conID string, operations []RepoOperation) ([]SubRes
 
 	return subResponsesFromBatchOperation, nil
 }
+
+// HTTPRequestWithRetryOnLock : Retries a request until either the maxHTTPRetries limit has been hit or the response status code is not equal to 423 (templates locked HTTP code)
+func HTTPRequestWithRetryOnLock(httpClient utils.HTTPClient, originalRequest *http.Request, connection *connections.Connection) (*http.Response, *sechttp.HTTPSecError) {
```

Comment:
as this is a useful utility function with apiroutes, should it take maxRetries as a param?

Reply:
It could be enhanced to be this way but at the moment its set at 5 as that is what Toby and I agreed on, if its going to be used as a utility param in the future and we need more or less than 5 then yes it will need to be changed.